src/blueapi/core/context.py

```diff
@@ -106,9 +106,15 @@ def with_device_module(self, module: ModuleType) -> None:
     def with_dodal_module(self, module: ModuleType, **kwargs) -> None:
         from dodal.utils import make_all_devices
 
-        for device in make_all_devices(module, **kwargs).values():
+        devices, exceptions = make_all_devices(module, **kwargs)
+
+        for device in devices.values():
             self.device(device)
 
+        num_failed = len(exceptions)
```

Comment:
Should we register devices that had exceptions?

Reply:
Has the signature of make_all_devices changed? Seems it used to return `dict[str, Device]` but now returns `tuple[dict[str, Device], dict[str, Exception]]`?

Reply:
It changes in the corresponding dodal PR, see description: https://github.com/DiamondLightSource/dodal/pull/550